inc/class-tmsc-custom-landing-page-types.php

```diff
@@ -0,0 +1,816 @@
+			if ( 'single' === $this->current_landing_type || ! empty( $this->current_term->term_id ) ) {
+				$wp_post_types[ $this->current_post_type ]->labels->add_new = $this->types[ $this->current_taxonomy ]['labels']['add_new_child'];
+				$wp_post_types[ $this->current_post_type ]->labels->add_new_item = $this->types[ $this->current_taxonomy ]['labels']['add_new_child'];
+				$wp_post_types[ $this->current_post_type ]->labels->edt_item = $this->types[ $this->current_taxonomy ]['labels']['edit_child'];
+				$wp_post_types[ $this->current_post_type ]->label = $this->types[ $this->current_taxonomy ]['labels']['child_name'];
+
+				// Override the parent labels with the child labels.
+				if ( ! empty( $this->current_term->term_id ) ) {
+					// Edit the title to use the term name for child pages
+					$edit_title = str_replace( $this->types[ $this->current_taxonomy ]['labels']['singular_name'], $this->current_term->name, $this->types[ $this->current_taxonomy ]['labels']['child_name'] );
+
+					$wp_post_types[ $this->current_post_type ]->labels->name = $edit_title;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Add in our custom colums.
+	 */
+	public function custom_columns( $column, $post_id ) {
+	}
+
+	/**
+	 * Hide the landing page info metabox on child posts. This can be disabled using the `tmsc_hide_child_custom_landing_pages` filter hook.
+	 * @param array $hidden. Array of element ids to be hidden.
+	 * @param object $screen. Screen to be hidden on.
+	 * @param boolean $use_defaults.
+	 * @return array.
+	 */
+	public function hide_landing_page_info_metabox( $hidden, $screen, $use_defaults ) {
+		if ( $this->on_custom_landing_admin_page() && ! empty( $this->current_landing_type ) && 'single' === $this->current_landing_type ) {
+			$hidden[] = '';
+		}
+		return $hidden;
+	}
+
+	/**
+	 * Add in an icon to display the current landing page type.
+	 */
+	public function add_landing_type_meta_box_info( $post ){
+
+		// If the post has been created, we can grab the saved info.
+		if ( $this->on_custom_landing_admin_page( $post ) && ! empty( $this->current_landing_type ) && 'single' === $this->current_landing_type ) {
+			$name = $this->types[ $this->current_taxonomy ]['labels']['singular_name'];
+			$terms = get_terms( array( 'taxonomy' => $this->current_taxonomy, 'hide_empty' => false ) ); ?>
+			<div class="misc-pub-section misc-pub-lptype">
+				<span id="custom-landing-page-pub-section">
+					<span class="dashicons <?php echo esc_attr( $this->types[ $this->current_taxonomy ]['menu_icon']); ?>"></span>
+				</span>
+				&nbsp;<?php echo esc_html( $name ); ?>: <b><span class="ai-target-text"><?php echo ( ! empty( $this->current_term ) ) ? esc_html( $this->current_term->name ) : ''; ?></span></b>
+				<a href="#custom-landing-page-type" class="ai-edit-button edit-custom-landing-page-type hide-if-no-js" role="button">
+					<span aria-hidden="true"><?php _e( 'Edit', 'ai' ); ?></span>
```

Comment:
Even though this is just static text I would still escape it with `esc_html_e` to future proof it.